File: 012.c
#include <stdio.h>

int divisors(long long int num){
  long long int lim = num/2 + 1;
  int count = 0;
  for (long long int i = 1; i < lim; i++) {
    if (num % i == 0) count++;
  }
  return count;
}

int main(){
  long long int t = 0;
  int count = 1;

  int num_div = 0;

  while (1) {
    t += (long long int)count++;
    if (divisors(t) > 500) { break; }
  }

  printf("triangle number: %lld\n", t);
  
  return 0;
}

File: 012.py

def getDivisors(num):
    ans = []
    for i in range(1, num/2 + 1):
        if num % i == 0:
            ans.append(i)
    return ans


t = 0
count = 1

while True:
    t += count
    count += 1

    if len(getDivisors(t)) > 500:
        print t
        exit()

